configurable resource names for infra VMs

vgamayunov opened the pull request:
Resource names configurable via config.yml for the following infra VMs:
- ccportal
- scheduler
- ondemand

xpillons commented:
what about the other VMs ? Jumpbox, grafana and deployer ?

vgamayunov commented:
> what about the other VMs ? Jumpbox, grafana and deployer ?

not implemented yet. these 3 are enough to satisfy the customer requirement. do you think we should do them all in one PR? I'm ok with that

xpillons commented:
> > what about the other VMs ? Jumpbox, grafana and deployer ?
> 
> not implemented yet. these 3 are enough to satisfy the customer requirement. do you think we should do them all in one PR? I'm ok with that

yes, otherwise it will stay in backlog forever ;-)

xpillons commented on tf/templates/global_variables.tmpl:
```diff
@@ -26,3 +26,4 @@ dns_ruleset_name              : ${dns-ruleset-name}
 scheduler_name                : ${scheduler-name}
```
I think this can be simplified because the VM names are not an output of the deployment process but are defined in the config.yml file. So I would remove all these from the global template file, and instead read these values from the config file. Ansible playbooks loads the config file as global variables so it should be straightforward to use this technics.
Doing it will also avoid the need to redeploy in order to rebuild the all.yml file on an existing environment.

vgamayunov replied:
True, however all these values are optional, and this will add more logic to the ansible playbooks. How do we deal with the default values so that they don't appear in many places?

xpillons replied:
Best would be to use this default syntax
"{{ ondemand.name | default('ondemand') }}"

vgamayunov commented:
added 3 more configurable VMs. So the list is now:
- scheduler
- ondemand
- ccportal
- grafana
- jumpbox
- deployer

also moved vm names from global_variables to ansible (with defaults).
works on the test cluster with all names modified, needs good testing now

xpillons commented on bicep/azhop.bicep:
```diff
@@ -378,6 +382,7 @@ var config = {
     deployJumpbox ? {
       jumpbox: {
         subnet: 'frontend'
+        name: contains(azhopConfig.jumpbox, 'name') ? azhopConfig.jumpbox.name : 'jumpbox'
```
jumpbox is optional, please add contains(azhopConfig, 'jumpbox')

xpillons commented on bicep/azhop.bicep:
```diff
@@ -390,6 +395,7 @@ var config = {
     deployGrafana ? {
       grafana: {
         subnet: 'admin'
+        name: contains(azhopConfig.grafana, 'name') ? azhopConfig.grafana.name : 'grafana'
```
grafana is optional, please add contains(azhopConfig, 'grafana')

ltalirz commented:
Thanks a lot for this @vgamayunov , this is also very useful for me!

I notice the name of the `ad` VM remains hardcoded, is there a particular reason for this?
